src/jaxoplanet/experimental/starry/flux.py

```diff
@@ -0,0 +1,79 @@
+import jax.numpy as jnp
+import numpy as np
+
+from jaxoplanet.experimental.starry.basis import A1, U0, A2_inv
+from jaxoplanet.experimental.starry.pijk import Pijk
+from jaxoplanet.experimental.starry.rotation import left_project
+from jaxoplanet.experimental.starry.solution import solution_vector
+
+
+# TODO: figure out the sparse matrices (and Pijk) to avoid todense()
+def flux(deg, theta, xo, yo, zo, ro, inc, obl, y, u, f):
+    u_deg = len(u)
+    U = jnp.array([1, *u])
+    full_deg = deg + u_deg
+    b = jnp.sqrt(jnp.square(xo) + jnp.square(yo))
+    b_rot = jnp.logical_or(jnp.greater_equal(b, 1.0 + ro), jnp.less_equal(zo, 0.0))
+    b_occ = jnp.logical_not(b_rot)
+
+    # Occultation
+    theta_z = jnp.arctan2(xo, yo)
+    sT = solution_vector(full_deg)(b, ro)
+    sTA2 = jnp.linalg.solve(A2_inv(full_deg).T.todense(), sT.T).T
```

Comment:
Here we might be better off using something like:

```suggestion
    A2 = scipy.sparse.linalg.inv(A2_inv(full_deg))
    A2 = jax.experimental.sparse.BCOO.from_scipy_sparse(A2)
    sTA2 = sT @ A2
```

(Note: I haven't actually tested this, but it does seem like `A2` is sparse.)